Added a user manual and a readme. (#2)
spatialos/UnrealGDKTestGyms: USER_MANUAL.md

# SpatialOS GDK for Unreal Test Gyms Project 

#### What is a gym
* A gym is a level containing a minimal number of actors, with simple behaviours, necessary to demonstrate some peice of functionality in the Spatial GDK for Unreal.
* We use gyms to help quickly visualize if some feature is working.

#### Project layout
```
Content
|
|__Actors
|
|__Characters
|
|__GameModes
|
|__Maps
|
|__Meshes
|
|__Spatial
```

New gym levels should be stored in `Content\Maps`.
New actors to be used in gyms should be stored in `Contetnt\Actors`.

#### How to add a gym 
* Create a new level, with a descriptive name, and store it in the `Content\Maps` directory;
  * consider copying the `EmptyGym` level in `Content\Maps` as a starting point.
* Consider adding a text object containing the gym name to the level;
  * this aids indentification from screen shots etc.
* Populate the new level with actors needed to demonstrate the desired functionality;
  * prefer to use existing actors if possible,

>   * if new actors are needed then store them in `Content\Actors`.

Reviewer: Do we want to have a guideline that all new gyms should have a README or other doc that documents the functionality tested in the gym?

Otherwise, LGTM.

Author: That sounds like something we should do. Will add a section underneath this and add the handover gym and the empty gym to it.

# SpatialOS GDK for Unreal Test Gyms Project 

#### What is a gym
*A gym is a level containing a minimal number of actors, with simple behaviours, necessary to demonstrate some peice of functionality in the Spatial GDK for Unreal.

> *We use gyms to helps quickly, visualy identifiy if some feature is working.

Reviewer 2: ```suggestion
*We use gyms to help quickly visualize if some feature is working.
```

#### Project layout
```
Content
|
|__Actors
|
|__Characters
|
|__GameModes
|
|__Geometry
|
|__Maps
|

> |__Meshes

Reviewer 2: What's the difference between Geometry and Meshes?

Author: They are just folders we didn't delete. So whatever the blank project puts in them. Presumably one is physics and one is rendering?

Author: Discussed offline. Will get rid of the geometry folder

|
|__Spatial
```

> New gym levels should be store in `Content\Maps`.

Reviewer 2: ```suggestion
New gym levels should be stored in `Content\Maps`.
```

# SpatialOS GDK for Unreal Test Gyms Project 

#### What is a gym

> * A gym is a level containing a minimal number of actors, with simple behaviours, necessary to demonstrate some peice of functionality in the Spatial GDK for Unreal.

Author: I thought we used the uk spelling for behaviour because unity did that so it's in our docs. Either way I'll commit this as suggested for now and we can fix that later if needed.

* We use gyms to help quickly visualize if some feature is working.

#### Project layout
```
Content
|
|__Actors
|
|__Characters
|
|__GameModes
|
|__Maps
|
|__Meshes
|
|__Spatial
```

New gym levels should be stored in `Content\Maps`.
New actors to be used in gyms should be stored in `Contetnt\Actors`.

#### How to add a gym 
1. Create a new level, with a descriptive name, and store it in the `Content\Maps` directory;
  * consider copying the `EmptyGym` level in `Content\Maps` as a starting point.
2. Consider adding a text object containing the gym name to the level;
  * this aids indentification from screen shots etc.

> 3. Populate the new level with actors needed to demonstrate the desired functionality;

Reviewer 3: ```suggestion
3. Populate the new level with the Actors required to demonstrate the functionality of this gym.
```
Slight rejig to make the sentence flow.

>   * prefer to use existing actors if possible,

Reviewer 3: ```suggestion
  * Use existing Actors in the Test Gyms Project if possible, in order to keep it minimal.
```

>   * if new actors are needed then store them in `Content\Actors`.

Reviewer 3: ```suggestion
  * If you create new Actors, store them in the `Content\Actors` directory.
```

> 4. Add a description of the gym to this document;

Reviewer 3: ```suggestion
4. Add a description of the new gym level to this `USER_MANUAL.md` document, in the section below;
```

>   * breifly describe what it tests and how to use it. 

Reviewer 3: ```suggestion
  * breifly describe what it demonstrates and how to use it. 
```

We said at the top the Test Gyms project is to demonstrate, so using consistent term in the doc.

#### Current gyms

##### Empty gym

> * This is used as a template for creating new gyms.

Reviewer 3: ```suggestion
* The template for creating new gyms. Copy this to use as a starting point for your own gym.
```

Made it more specific.

##### Handover gym

> * Used to test that an entity can cross from one load balancing region to another.

Reviewer 3: Should we keep with "demonstrates"?
We call load-balancing regions "areas of authority" in the docs. I presume this is what this means?

```suggestion
* Demonstrates that an entity can cross from one area of authority to another.
```

* Contains a cube that moves back and forth across a floor.

-----

> 2019-11-06: Page added with limited editorial review

Reviewer 3: ```suggestion
2019-11-06: Page added with editorial review
```

Reviewer 3: Upgraded the doc editorial status.
